sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -279,6 +269,50 @@ case class Ceil(child: Expression) extends UnaryMathExpression(math.ceil, "CEIL"
   override protected def withNewChildInternal(newChild: Expression): Ceil = copy(child = newChild)
 }
 
+@ExpressionDescription(
+  usage = "_FUNC_(expr) - Returns the smallest integer not smaller than `expr`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(-0.1);
+       0
+      > SELECT _FUNC_(5);
+       5
+      > SELECT _FUNC_(3.1411, 3);
+       3.142
+      > SELECT _FUNC_(3.1411, -3);
+       1000
+  """,
+  since = "3.3.0",
+  group = "math_funcs")
+object CeilExpressionBuilder extends ExpressionBuilder {
+  def build(expressions: Seq[Expression]): Expression = {
+    if (expressions.length == 1) Ceil(expressions.head)
+    else if (expressions.length == 2) {
+      val child = expressions(0)
+      val scale = expressions(1)
+      val scaleV = scale.eval(EmptyRow)
+      if (scaleV == null) throw new AnalysisException("Scale parameter can not be null")
+
+      (child.dataType, scaleV.asInstanceOf[Int]) match {
```

Comment:
ditto, we should check `scale.dataType` first, otherwise we may hit cast error when doing `asInstanceOf`